### DISCUSSION TITLE
Why lock on dataclass?

### DISCUSSION THREAD

**Discussion by lev-trullion** [oxan/djangorestframework-dataclasses#74] [General]
Why lock on a dataclass? Why not just any class?

**Comment by oxan** [comment]
I've converted this to an discussion, as it's a question and not an actionable issue.

The short answer is because this package uses various features from the `dataclasses` module (mostly `dataclasses.fields` and all its metadata processing, but also `dataclasses.replace`). The longer answer is that there haven't been any requests for regular class support nor do I see it having much added value, and it does require real work in reimplementing all the features from the dataclasses module.

**Comment by lev-trullion** [comment]
I'd like same library for attrs lib
